tests/host_tools/network.py

```diff
@@ -52,7 +54,17 @@ def __init__(self, netns, ssh_key: Path, host, user):
             str(self.ssh_key),
         ]
 
-        self._init_connection()
+        # _init_connection loops until it can connect to the guest
+        # dumping debug state on every iteration is not useful or wanted, so
+        # only dump it once if _all_ iterations fail.
+        try:
+            self._init_connection()
+        except Exception as exc:
+            on_error(exc)
```

Comment:
Shouldn't this be already be handled in `_exec`, so we are handled it here a 2nd time?

Reply:
For `_init_connection`, inside of `_exec` the `_on_error` field is still `None` because the constructor only sets it after `_init_connection`. This is to avoid logging a ton of stuff on every retry.